validator.js

```diff
@@ -141,6 +141,23 @@ var validator = function(options) {
 
   // Load schemas from folder
   if (options.folder) {
+
+    // Load JSON schema for JSON Schemas
+    var coreSchema = fs.readFileSync(
+      path.join(__dirname, 'schemas', 'draft-04-schema.json'), {
+      encoding: 'utf-8'
+    });
+    // Create validator for core schema, so we can report schemas that are
+    // invalid
+    var coreSchemaValidator = new jjv();
+    coreSchemaValidator.addSchema(JSON.parse(coreSchema));
+    var validateSchema = function(schemaToValidate) {
+      return coreSchemaValidator.validate(
+        'http://json-schema.org/draft-04/schema#',
+        schemaToValidate
+      );
```

Comment:
Yeah... all the schema we load over network are declared upfront... it would have to a major change...
Besides the function is synchronous, so it's very unlikely that it does I/O.